### PR TITLE
feat: add license mismatch detection for pyproject.toml

### PR DESCRIPTION
**Pull request by Strum355**
## Summary
- Implements `extractLicensePosition()` in the pyproject.toml provider to enable license mismatch highlighting for Python projects
- Supports three license field formats:
  - Poetry style: `[tool.poetry] license = "MIT"`
  - PEP 639: `[project] license = "MIT"` (simple string)
  - PEP 621: `[project] license = {text = "MIT"}` (inline table)
- `[project]` takes precedence over `[tool.poetry]` when both are present

## Test plan
- [x] 7 new test cases covering all formats, edge cases, and precedence
- [x] All 289 tests pass (96.8% statement coverage for pyproject.toml.ts)
- [ ] Manual verification: open a pyproject.toml with a license mismatch and confirm the license field is underlined
- [ ] Manual verification: apply quick-fix and confirm the replacement is correct

Ref: TC-4697

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Add license field position extraction for pyproject.toml to enable license mismatch detection in Python projects.

New Features:
- Support extracting license positions from [project] license string fields (PEP 639) and inline table license fields with text keys (PEP 621) in pyproject.toml.
- Support extracting license positions from [tool.poetry] license fields in pyproject.toml, with [project] entries taking precedence when both exist.

Tests:
- Add unit test suite for pyproject.toml license position extraction covering multiple formats, invalid TOML, missing license fields, and precedence rules.

### CODE REVIEW

File: test/providers/license.pyproject.toml.test.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+license = "BSD-3-Clause"
+
+[tool.poetry]
+license = "MIT"
+`;
+        const result = dependencyProvider.extractLicensePosition?.(contents);
+
+        expect(result).to.not.be.undefined;
+        expect(result?.value).to.equal('BSD-3-Clause');
+    });
```
[the remark this answers is not in the document]
Comment (Strum355) [reply test/providers/license.pyproject.toml.test.ts:119]: Good catch — added position assertion to the existing precedence test and a new reversed-order test case in fb0c2f9.

File: src/providers/pyproject.toml.ts [hunk 2]
```diff
@@ -200,4 +200,64 @@ export class DependencyProvider extends EcosystemDependencyResolver implements I
...
+        // [tool.poetry] — Poetry-style license
+        if (this.keyPathEquals(node.resolvedKey, 'tool', 'poetry')) {
+          const licenseKv = node.body.find(kv => keyName(kv.key.keys[0]) === 'license');
+          if (licenseKv && licenseKv.value.type === 'TOMLValue' && licenseKv.value.kind === 'string') {
+            return {
+              value: String(licenseKv.value.value),
+              position: this.toValuePosition(licenseKv.value.loc),
+            };
+          }
+        }
```
[the remark this answers is not in the document]
Comment (Strum355) [reply src/providers/pyproject.toml.ts:258]: Valid find — [tool.poetry] now stores a fallback instead of returning immediately, so [project] always takes precedence per PEP 621 regardless of section order. Added a reversed-order regression test. Fixed in fb0c2f9.

**Comment by Strum355** [comment]
Re Sourcery's high-level feedback:

- **"`ast.body[0].body` misses later tables"** — incorrect. `toml-eslint-parser` puts all tables under a single `TOMLTopLevelTable` at `ast.body[0]`. This is the same pattern used by `mapDependencies()` in this file. Verified independently.
- **"guard `kv.key.keys[0]` against empty array"** — unnecessary. A parsed TOML key-value node always has at least one key element. Same access pattern used throughout the file without guards.

The precedence bug and test suggestion were both valid and addressed in fb0c2f9.

**Review by a-oren — APPROVED** [review]
LGTM
